File: apps/desktop/src/contracts.ts
export type ComponentState = "ready" | "configured" | "unavailable" | "unsupported"

export interface ComponentHealth {
  state: ComponentState
  version?: string
  detail: string
}

export interface DesktopHealth {
  appVersion: string
  ready: boolean
  opencode: ComponentHealth
  sandbox: ComponentHealth
  runtimeTools: ComponentHealth
  modelEndpoint: ComponentHealth
}

export interface SelectedDocument {
  selectionId: string
  displayName: string
}

export interface WorkflowReceipt {
  jobId: string
}

export const DESKTOP_HEALTH_EVENT = "desktop://health"

File: apps/desktop/src-tauri/src/contracts.rs
use serde::Serialize;

pub const DESKTOP_HEALTH_EVENT: &str = "desktop://health";

#[derive(Clone, Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ComponentHealth {
    pub state: ComponentState,
    pub version: Option<String>,
    pub detail: String,
}

#[derive(Clone, Copy, Debug, Serialize)]
#[serde(rename_all = "snake_case")]
pub enum ComponentState {
    Ready,
    Configured,
    Unavailable,
    Unsupported,
}

#[derive(Clone, Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DesktopHealth {
    pub app_version: String,
    pub ready: bool,
    pub opencode: ComponentHealth,
    pub sandbox: ComponentHealth,
    pub model_endpoint: ComponentHealth,
}
